src/orion/storage/track.py

```diff
@@ -0,0 +1,464 @@
+====================================================
+
+.. module:: base
+   :platform: Unix
+   :synopsis: Implement a storage protocol to allow Orion to use track as a storage method
+
+"""
+
+from collections import defaultdict
+import datetime
+import logging
+import uuid
+
+try:
+    from track.client import TrackClient
+    from track.persistence.utils import parse_uri
+    from track.serialization import to_json
+    from track.structure import CustomStatus, Status as TrackStatus
+    from track.structure import Project, Trial as TrackTrial, TrialGroup
+
+    HAS_TRACK = True
+except ImportError:
+    HAS_TRACK = False
+
+from orion.core.worker.trial import Trial as OrionTrial
+from orion.storage.base import BaseStorageProtocol
+
+
+log = logging.getLogger(__name__)
+
+
+if HAS_TRACK:
+    _status = [
+        CustomStatus('new', TrackStatus.CreatedGroup.value + 1),
+        CustomStatus('reserved', TrackStatus.CreatedGroup.value + 2),
+
+        CustomStatus('suspended', TrackStatus.FinishedGroup.value + 1),
+        # CustomStatus('completed', TrackStatus.FinishedGroup.value + 2),
+
+        CustomStatus('interrupted', TrackStatus.ErrorGroup.value + 1),
+        CustomStatus('broken', TrackStatus.ErrorGroup.value + 3)
+    ]
+
+    _status_dict = {
+        s.name: s for s in _status
+    }
+    _status_dict['completed'] = TrackStatus.Completed
+
+
+def get_track_status(val):
+    """Convert orion status to track status"""
+    return _status_dict.get(val)
+
+
+def convert_track_status(status):
+    """Convert track status to orion status"""
+    return status.name.lower()
+
+
+def remove_leading_slash(name):
+    """Remove leading slash"""
+    # if name[0] == '/':
+    #     return name[1:]
+    # return name
+    return name
+
+
+def add_leading_slash(name):
+    """Add leading slash"""
+    # if name[0] == '/':
+    #     return name
+    # return '/' + name
+    return name
+
+
+class TrialAdapter:
+    """Mock Trial, see `~orion.core.worker.trial.Trial`
+
+    Parameters
+    ----------
+    storage_trial
+        Track trial object
+
+    orion_trial
+        Orion trial object
+
+    objective: str
+        objective key
+
+    """
+
+    def __init__(self, storage_trial, orion_trial=None, objective=None):
+        self.storage = storage_trial
+        self.memory = orion_trial
+        self.session_group = None
+        self._params = None
+        self.objective_key = objective
+        self.objectives_values = None
+        self._results = []
+
+    def _repr_values(self, values, sep=','):
+        """Represent with a string the given values."""
+        return
+
+    def __str__(self):
+        """Represent partially with a string."""
+        param_rep = ','.join(map(lambda value: "{0.name}:{0.value}".format(value), self.params))
+        ret = "TrialAdapter(uid={3}, experiment={0}, status={1}, params={2})".format(
+            repr(self.experiment[:10]), repr(self.status), param_rep, self.storage.uid)
+        return ret
```

Comment:
Is this string supposed to be different than `str(Trial())`?

Reply:
yes, more compact & easier to read

Reply:
I would suggest we harmonize str(Trial) and str(TrialAdapter) for coherence. Do you think it is worth changing str(Trial)?